frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/SlideInDialog.kt

```diff
@@ -1,129 +1,274 @@
-  var branchName = ""
-  var reattach = false
-  var reattachCheckbox: JCheckBox? = null
   val rootNames = branchLayout.rootEntries.map { it.name }
 
+  var unmergedBranches = emptyList<String>()
+
+  val reattachCheckbox =
+    JCheckBox(
+      getString(
+        "action.GitMachete.BaseSlideInBelowAction.dialog.slide-in.checkbox.reattach"
+      )
+    )
+
+  val branchField = createBranchField()
+  val innerPanel = createInnerPanel()
+  val panel = createPanel()
+
   init {
+    loadUnmergedBranchesInBackground()
     title = getString("action.GitMachete.BaseSlideInBelowAction.dialog.slide-in.title")
     setOKButtonText(
       getString("action.GitMachete.BaseSlideInBelowAction.dialog.slide-in.ok-button")
     )
+    updateBranchesField()
     setOKButtonMnemonic('S'.code)
-    super.init()
+    init()
+    rerender()
   }
 
-  fun showAndGetBranchName() =
-    if (showAndGet()) SlideInOptions(branchName.trim(), reattach) else null
+  override fun createCenterPanel() = panel
 
-  override fun createCenterPanel() = panel {
-    row(getString("action.GitMachete.BaseSlideInBelowAction.dialog.slide-in.label.parent")) {
-      label(parentName).bold()
-    }
-    row {
-      label(
-        getString(
-          "action.GitMachete.BaseSlideInBelowAction.dialog.slide-in.label.branch-name"
-        )
+  override fun getPreferredFocusedComponent() = branchField
+
+  override fun doValidateAll(): List<ValidationInfo> =
+    listOf(::validateBranchName).mapNotNull { it() }
+
+  fun getSlideInOptions(): SlideInOptions {
+    val branchName = branchField.getText().orEmpty().trim()
+    return SlideInOptions(branchName, reattachCheckbox.isSelected)
+  }
+
+  private fun loadUnmergedBranchesInBackground() =
+    ProgressManager.getInstance()
+      .run(
+        object :
+          Task.Backgroundable(
+            project,
+            getString(
+              "action.GitMachete.BaseSlideInBelowAction.dialog.slide-in.progress"
+            ),
+            true
+          ) {
+
+          @UIThreadUnsafe
+          override fun run(indicator: ProgressIndicator) {
+            val root = (gitRepository.root)
+            loadUnmergedBranchesForRoot(root)?.let { branches -> unmergedBranches = branches }
+          }
+        }
       )
-    }
-    row {
-      textField().bindText(::branchName).validationOnApply(validateBranchName()).focused().apply {
-        startTrackingValidationIfNeeded()
+
+  /**
+   * ```
+   * $ git branch --all
+   * |  master
+   * |  feature
+   * |* checked-out
+   * |+ checked-out-by-worktree
+   * |  remotes/origin/master
+   * |  remotes/origin/feature
+   * |  remotes/origin/HEAD -> origin/master
+   * ```
+   */
+  @UIThreadUnsafe
+  private fun loadUnmergedBranchesForRoot(root: VirtualFile): List<String>? {
+    var result: List<String>? = null
+
+    val handler =
+      GitLineHandler(project, root, GitCommand.BRANCH).apply {
+        addParameters("--no-color", "-a", "--no-merged")
```

Comment:
Hmmm `git branch` is still a **porcelain** command... I'd rather resort to the underlying **plumbing** command, `git for-each-ref`. See `git machete status -v` for an example invocation

Reply:
actually `unmergedBranches` is not used so the whole loading is redundant

Reply:
Wow okay... why was it used originally but isn't anymore? 🤔

Reply:
This class is based on the other IntelliJ dialog. I was removing some redundant parts and tweaking the others - left accidentally